sdk/identity/identity/src/credentials/azurePipelinesCredential.ts

```diff
@@ -171,11 +185,17 @@ export function handleOidcResponse(response: PipelineResponse): string {
     }
   } catch (e: any) {
     const errorDetails = `${credentialName}: Authentication Failed. oidcToken field not detected in the response.`;
-    logger.error(`Response from service = ${text} and error message = ${e.message}`);
+    logger.error(
+      `Response from service = ${text}, Headers - ${JSON.stringify(
+        response.headers
```

Comment:
Headers in the allow list will be shown here. They won't be REDACTED since they are part of the allow list. Did I answer your question?

Reply:
I believe calling `logger.error` will not redact anything by default does it? it's the `sanitizer` that does the clean up of headers https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/src/policies/logPolicy.ts#L58

So if we don't want to log certain headers we probably need to either:
1. log only the values we care about / headers we want to see
2. Use the sanitizer

I think (2) is much harder, because I dont think our [sanitizer](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/src/util/sanitizer.ts) is exported at the top level